src/redux/books/books.js

```diff
@@ -1,45 +1,96 @@
-// Actions
-const ADD = 'add';
-const DEL = 'del';
-
-// Empty array of books
-const initialState = [{
-  id: 1,
-  genre: 'genre',
-  title: 'book title',
-  author: 'author',
-  status: 'complete',
-  currentChapter: 'Current chapter',
-  chapterNumber: 'Chapter #number',
-},
-];
-
-// Reducer
-export default function booksReducer(state = initialState, action = {}) {
+/* eslint-disable camelcase */
+const URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/ewYk4YjZVq8pYtMAfktj/books';
+
+// action types
+const ADD = 'BOOK_ADDED';
+const REMOVE = 'BOOK_REMOVED';
+const READ = 'BOOKS_RETRIEVED';
+
+// Action creators
+export const addBook = ({
+  item_id, title, author, category,
+}) => ({
+  type: ADD,
+  item_id,
+  title,
+  author,
+  category,
+});
+
+export const removeBook = (item_id) => ({
+  type: REMOVE,
+  item_id,
+});
+
+export const readBooks = (books) => ({
+  type: READ,
+  books,
+});
+
+const getFromAction = ({
+  item_id, title, author, category,
+}) => ({
+  item_id, title, author, category,
+});
+
+const booksReducer = (state = [], action) => {
   switch (action.type) {
     case ADD:
-      return [...state, {
-        ...action.payload,
-      }];
-    case DEL:
-      return [...state.filter((book) => book.id !== action.payLoad.id)];
-    default: return state;
+      return [
+        ...state,
+        getFromAction(action),
+      ];
+    case REMOVE:
+      return state.filter((book) => book.item_id !== action.item_id);
+
+    case READ:
+      return action.books;
+
+    default:
+      return state;
   }
-}
-
-// Action Creators
-export function add(book) {
-  return {
-    type: ADD,
-    payload: book,
-  };
-}
-
-export function del(id) {
-  return {
-    type: DEL,
-    payLoad: {
-      id,
+};
+
+export const fetchBooks = () => async (dispatch) => {
+  await fetch(URL)
+    .then((res) => res.json())
+    .then((books) => {
+      const BookList = [];
+      Object.keys(books).forEach((key) => {
+        BookList.push({
+          item_id: key,
+          title: books[key][0].title,
+          author: books[key][0].author,
+          category: books[key][0].category,
+        });
+      });
+      dispatch(readBooks(BookList));
+    });
+};
+
+export const postBook = (book) => async (dispatch) => {
+  await fetch(URL, {
+    method: 'POST',
+    body: JSON.stringify(book),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
     },
-  };
-}
+  })
+    .then(() => {
+      dispatch(addBook(book));
+    });
+};
+
+export const deleteBook = (id) => async (dispatch) => {
+  await fetch(`${URL}/${id}`, {
+    method: 'DELETE',
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  })
+    .then(() => {
+      dispatch(removeBook(id));
+    });
+};
+
+export default booksReducer;
```

Comment:
- [ ] It is part of the project requirements to use either `fetch` or `axios` with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls. I mentioned this in my first review and instead of you to check the link to the documentation on how to use `createAsyncThunk`, you went ahead to create a new branch. Kindly take your time to study how to use it and implement the knowledge gained in your project.

![project-rqts](https://user-images.githubusercontent.com/26597432/214961727-a0a629c9-9caf-450f-9dff-8586be690fb7.png)